server/handlers.go

```diff
@@ -716,6 +721,30 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 		return
 	}
 
+	// Check for code challenge validity
+	if authCode.CodeChallenge != "" {
+		if verifier == "" {
+			s.tokenErrHelper(w, errInvalidRequest, "Require code_verifier", http.StatusUnauthorized)
+			return
+		}
+		var challenge string
+		switch authCode.CodeChallengeMethod {
+		case codeChallengeSHA256:
+			sum := sha256.Sum256([]byte(verifier))
+			challenge = base64.RawURLEncoding.EncodeToString(sum[:])
+		// default to plain: insecure
```

Comment:
Defaulting to something insecure... Anything else we could do?

Reply:
It's the default in the RFC, if the client wants S256, it has to specify it. See section 4.3